File: LeetCode/cpp/24.cpp
class Solution {
public:
    ListNode *swapPairs(ListNode *head) {
        if (!head || !head->next)
            return head;
        ListNode *first = head, *second = head->next, *ret = second;
        while (first && second) {
            if (second->next && !second->next->next) {
                first->next = second->next;
                second->next = first;
                break;
            }
            if (second->next)
                first->next = second->next->next;
            else
                first->next = nullptr;
            ListNode *temp = second->next;
            second->next = first;
            first = temp;
            if (first)
                second = first->next;
        }
        return ret;
    }
};

File: LeetCode/python3/24.py
class Solution:
    def swapPairs(self, head: ListNode) -> ListNode:
        if not head or not head.next:
            return head
        first, second = head, head.next if head else None
        head = second
        while first and second:
            cache = second.next
            second.next = first
            first.next = cache.next if cache and cache.next else cache
            first = cache
            second = first.next if first else None
        return head
    